### PR TITLE
Make _filter_columns robust to different encodings

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
For BH tutorial.

### CODE REVIEW

File: nilearn/datasets/utils.py [hunk 1]
```diff
@@ -408,6 +407,10 @@ def _filter_column(array, col, criteria):
         filter = array[col] <= criteria[1]
         return np.logical_and(filter, array[col] >= criteria[0])
 
+    # Handle strings with different encodings
+    if isinstance(criteria, (_basestring, bytes)):
+        criteria = np.array(criteria).astype(array[col].dtype.type)
```
Comment (GaelVaroquaux) [inline nilearn/datasets/utils.py]: Why not "astype(array[col].dtype)"? Why the extra ".type"?

**Comment by GaelVaroquaux** [comment]
I am waiting for this PR, and then cutting a release?

**Comment by AlexandreAbraham** [comment]
Fixed! Go on for the release, or I can take care of it if you want.

**Comment by AlexandreAbraham** [comment]
Scipy is back ! I'm waiting for CircleCI to go green and I do the release. @GaelVaroquaux, do not work on it anymore, I'm taking care.

**Comment by GaelVaroquaux** [comment]
OK, go for the release.

Don't forget to update the whats_new, and to add an entry in the sidebar.

**Pull request merged**